imap-codec/src/codec/encode.rs

```diff
@@ -154,6 +156,7 @@ impl Iterator for Encoded {
 }
 
 /// The intended action of a client or server.
+#[cfg_attr(feature = "serde", derive(Serialize))]
```

Comment:
Ah, good catch :-)

Reply:
Ehm, what did I catch? 😕

Reply:
Ah, nothing special. Just wondered if we forgot at some point to derive serde. Should we also already derive Deseralize here?

Reply:
About `Deserialize` for `Fragment`: I did not add it intentionally because I did not need it and could not see any public interface that takes a `Fragment`, thus why would any library user want to deserialize one? Sure, we can add it for symmetry - but then we would also need to add a test to cover the implementation. :wink:

About "we forgot": Currently, nothing in `imap-codec` does implement `Deserialize`/`Serialize`. Do we want all public types to be able to implement them? Then, we should create an issue for adding these together with appropriate unit tests.

Reply:
I thought about symmetry mainly. Having both pieces allows to write tests of the form `assert_eq!(x, deserialize(serialize(x))` and often comes in handy. But: we'll learn when we need it. I'll let you decide to add or not :-)

Reply:
I now added `Deserialize` as well for symmetry to prevent any future confusion.

After implementing Python classes for `Fragment` variants (`LineFragment`, `LiteralFragment`) (which I will work on when I'm done with all of this PR's side quests :wink:), the Python bindings would (very likely) no longer require  `Fragment` to implement `Serialize` which could be removed again.

However, as removing the implementations lateron would be breaking, we should decide "Do we want all public types to to implement `Deserialize`/`Serialize` (given the `serde` feature)?" before the 2.0 release. -> #555